#! /bin/sh -e
## 01_defaults.debian.dpatch by Tollef Fog Heen <tfheen@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Don't spit out a traceback if you are giving a admin password
## DP: Closes #233161

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch ${2:+-d $2}}"

if [ $# -lt 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argum
ent"
                exit 1;;
esac

exit 0
@DPATCH@

--- mailman-2.1.4/Mailman/Cgi/options.py2004-02-17 10:56:16.000000000 +0800
+++ /usr/lib/mailman/Mailman/Cgi/options.py2004-02-17 10:59:04.000000000 +0800
@@ -119,8 +119,12 @@
         return
     # Sanity check the user, but only give the "no such member" error when
     # using public rosters, otherwise, we'll leak membership information.
-    if not mlist.isMember(user) and mlist.private_roster == 0:
-        doc.addError(_('No such member: %(safeuser)s.'))
+    if not mlist.isMember(user):
+        if mlist.private_roster == 0:
+            doc.addError(_('No such member: %(safeuser)s.'))
+        else:
+            # Prevent an exception when the site/list admin password is used
+            doc.addError(_('Authentication failed.'))
         loginpage(mlist, doc, None, language)
         print doc.Format()
         return
